enhancements/installer/gcp-customer-provided-subnets.md

```diff
@@ -0,0 +1,258 @@
+- Internal and external DNS entries (google_dns_record_set)
+
+- Service Accounts (google_service_account, google_project_iam_member)
+
+- Bootstrap ignition bucket (google_storage_bucket, google_storage_bucket_object, google_storage_object_signed_url)
+
+- Instances (google_compute_instance, google_compute_address)
+
+#### Changes to resources created by the installer
+
+- The MCS firewall rule currently limits by source_ranges of network_cidr, master_nat_ip, and worker_nat_ip. However, we have no way of knowing what the master_nat_ip or worker_nat_ip is. As a result, the MCS firewall rule will need to become more generic, thus available externally (0.0.0.0/0).
+
+#### API
+The install-config YAML will grow two properties on the [gcp platform structure][gcp-platform-structure]
+
+- ControlPlaneSubnet
+
+- ComputeSubnet
+
+The install-config wizard will add three prompts:
+
+- A Select prompt:
+  Message: VPC
+  Help: The VPC to be used for installation.
+  Default: new
+  Options: [“new”] + results of a [networks.list][networks-list] call.
+
+- A Select prompt:
+  Message: Control-plane subnet
+  Help: The subnet to be used for the control plane instances.
+  Options: results of a [subnetworks.list][subnetworks-list] call filtered by the selected VPC.
+  Only when: VPC is not “new”
+
+- A Select prompt:
+  Message: Compute subnet
+  Help: The subnet to be used for the compute instances.
+  Options: results of a [subnetworks.list][subnetworks-list] call filtered by the selected VPC.
+  Only when: VPC is not “new”
+
+#### Isolation between clusters
+
+The clusters will be isolated by firewall rules. Specifically, by tagging the instances with ${INFRA_ID}-master and ${INFRA_ID}-worker, and using those tags in the firewall rules to ensure only desired traffic is allowed. However, there are a few modifications required to make this absolute:
+
+- ICMP ingress.  We may be able to drop this completely.
+
+- TCP 22 ingress (SSH). The bootstrap instance  allows global access (0.0.0.0/0). This can probably stay the same as it is only temporary and of limited risk.  The control-plane and compute instances allow the network_cidr [Networking.MachineCIDR][networking-machinecidr]. This could be restricted to tags if determined the risk is sufficient. Using different ssh keys would help as well.
+
+- TCP 22623 ingress (MCS). The MCS firewall rule is to be opened up externally in order to remove the need for master_nat_ip and worker_nat_ip as inputs. Switching to using internal load balancers would remediate this.
+
+#### Destroy
+
+There should be no work required. For gcp, we filter on name containing {INFRA_ID}. As a result, destroy should only delete things that match this naming convention.
+
+#### CI
+
+CI jobs should set up a VPC, subnets, routing, and NAT using our [UPI flow][upi-flow] and launch the cluster-under-test into that infrastructure.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
```

Comment:
You want to rip this template stuff out of enhancements and replace it with your own content.  Maybe pull down some of the _Isolation between clusters_ stuff, since that's talking about risks?